pkgs/top-level/perl-packages.nix

```diff
@@ -14747,6 +14762,52 @@ let
     };
   };
 
+  PDL = buildPerlPackage rec {
+    pname = "PDL";
+    version = "2.019";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/C/CH/CHM/${pname}-${version}.tar.gz";
+      sha256 = "51a47beedb1dd5bb4b6f6441a0ad38d4c95d7cbf3bf6b1b4c9128732ea2d77fe";
+    };
+    patchPhase = ''
+      substituteInPlace perldl.conf \
+        --replace 'POSIX_THREADS_LIBS => undef' 'POSIX_THREADS_LIBS => "-L${pkgs.glibc.dev}/lib"' \
+        --replace 'POSIX_THREADS_INC  => undef' 'POSIX_THREADS_INC  => "-I${pkgs.glibc.dev}/include"' \
+        --replace 'WITH_MINUIT => undef' 'WITH_MINUIT => 0' \
+        --replace 'WITH_SLATEC => undef' 'WITH_SLATEC => 0' \
+        --replace 'WITH_HDF => undef' 'WITH_HDF => 0' \
+        --replace 'WITH_GD => undef' 'WITH_GD => 0' \
+        --replace 'WITH_PROJ => undef' 'WITH_PROJ => 0'
+
+      # This unit test attempts to write to HOME(/homeless-shelter) with "~".
+      # Patch it to write to /tmp instead.
+      substituteInPlace t/fits.t --replace '~' '/tmp'
```

Comment:
Attempted to build this PR (with https://github.com/NixOS/nixpkgs/pull/84600 merged), and one of the tests fail. Perhaps due to substituteInPlace unintentionally replacing ~ in some pattern matching operators.

Latest upstream version might not require this replace.

Relevant parts of the output:
```
$ nix-build -A perlPackages.PDL
[...]
t/fits.t ........................ 1/97 Unknown regexp modifier "/b" at t/fits.t line 324, at end of line
Unmatched ) in regex; marked by <-- HERE in m/tmp) <-- HERE / at t/fits.t line 324.
# Looks like your test exited with 255 just after 1.
t/fits.t ........................ Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 96/97 subtests
[...]
Test Summary Report
-------------------
[...]
t/fits.t                      (Wstat: 65280 Tests: 1 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 97 tests but ran 1.
[...]
Files=135, Tests=1536, 49 wallclock secs ( 0.59 usr  0.15 sys + 31.08 cusr 17.85 csys = 49.67 CPU)
Result: FAIL
Failed 1/135 test programs. 0/1536 subtests failed.
make: *** [Makefile:1103: test_dynamic] Error 255
builder for '/nix/store/4jj8ii0c8cw5lq5axap78c9gfn3hw5ml-perl5.30.1-PDL-2.019.drv' failed with exit code 2
e
```